### PR TITLE
[14.0][MIG] sale_margin_sync: Migration to 14.0

### PR DESCRIPTION
**Pull request by rafamarpe**

### CODE REVIEW

**Comment by rafamarpe** [comment]
@pedrobaeza it's possible to merge

**Review by pedrobaeza — APPROVED** [review]
Please enroll another of your colleagues to review the PR, or you can review other PRs and ask in exchange that they review yours.

**Comment by rafamarpe** [comment]
@HaraldPanten it's possible to merge.

**Comment by pedrobaeza** [comment]
@rafamarpe please check my previous comment about your colleagues or the reciprocity.

**Comment by rafamarpe** [comment]
Ok Thank's @pedrobaeza sorry!

**Comment by HaraldPanten** [comment]
> Ok Thank's @pedrobaeza sorry!

Hi @rafamarpe ,

Once we have other validations I'll be happy to merge this one (it's been validated already by Pedro). I don't have availability at the moment to check this one, sorry, but you can attend Pedro's comments in order to get more reviews 👍

**Review by pedroortega-punt — COMMENTED** [review]
Checked in client and runbot ok

**Review by pedroortega-punt — APPROVED** [review]
Checked in client and runbot ok

File: sale_margin_sync/models/stock_valuation_layer.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        if "unit_cost" in vals and not self.env.context.get(
+            "skip_sale_margin_sync", False
+        ):
+            self.sale_margin_sync()
+        return res
+
+    def sale_margin_sync(self):
+        """Only synchronize outgoing moves"""
+        for svl in self.filtered(lambda l: (l.quantity < 0.0)):
+            svl.stock_move_id.sale_line_id.purchase_price = svl.unit_cost
```
Comment (enriquemartin) [inline sale_margin_sync/models/stock_valuation_layer.py]: I think we need to take into account the unit of measure here. The code could be something like:
```
    def sale_margin_sync(self):
        """Only synchronize outgoing moves"""
        for svl in self.filtered(lambda l: (l.quantity < 0.0)):
            sale_line_id = svl.stock_move_id.sale_line_id
            product_cost = svl.unit_cost
            if sale_line_id.product_uom and sale_line_id.product_uom != svl.uom_id:
                product_cost = svl.uom_id._compute_price(
                    product_cost,
                    sale_line_id.product_uom,
                )
            sale_line_id.purchase_price = product_cost
```

**Review by enriquemartin — CHANGES_REQUESTED** [review]
I have added some comments in relation to possible different units of measure:

I have replicated the problem in runboat (I added the `stock.propagate_uom system` parameter with value 1 to propagate sale order line uom to stock move): order S00025 we sell 2 dozens. 
At first, the cost is correctly calculated, but when it is updated it is based on the unit of measure of the product. The correct cost should be 107,50€ x 12 = 1290,00€. However, it has transferred the cost of the unit.
Thanks

File: sale_margin_sync/models/stock_valuation_layer.py (hunk 1 above)
Comment (rafamarpe) [reply sale_margin_sync/models/stock_valuation_layer.py]: Ok I make the changes and add the tests, thanks

Comment (rafamarpe) [reply sale_margin_sync/models/stock_valuation_layer.py]: Changes made

**Review by enriquemartin — APPROVED** [review]

**Pull request merged**